examples/federated/core_package/package.json

```diff
@@ -8,106 +8,86 @@
+    "@jupyterlab/filebrowser": "~4.0.0-alpha.2",
+    "@jupyterlab/filebrowser-extension": "~4.0.0-alpha.2",
+    "@jupyterlab/fileeditor": "~4.0.0-alpha.2",
+    "@jupyterlab/fileeditor-extension": "~4.0.0-alpha.2",
+    "@jupyterlab/help-extension": "~4.0.0-alpha.2",
+    "@jupyterlab/htmlviewer-extension": "~4.0.0-alpha.2",
+    "@jupyterlab/hub-extension": "~4.0.0-alpha.2",
+    "@jupyterlab/imageviewer": "~4.0.0-alpha.2",
+    "@jupyterlab/imageviewer-extension": "~4.0.0-alpha.2",
+    "@jupyterlab/inspector": "~4.0.0-alpha.2",
+    "@jupyterlab/inspector-extension": "~4.0.0-alpha.2",
+    "@jupyterlab/javascript-extension": "~4.0.0-alpha.2",
+    "@jupyterlab/json-extension": "~4.0.0-alpha.2",
+    "@jupyterlab/launcher": "~4.0.0-alpha.2",
+    "@jupyterlab/launcher-extension": "~4.0.0-alpha.2",
+    "@jupyterlab/logconsole": "~4.0.0-alpha.2",
+    "@jupyterlab/logconsole-extension": "~4.0.0-alpha.2",
+    "@jupyterlab/mainmenu": "~4.0.0-alpha.2",
+    "@jupyterlab/mainmenu-extension": "~4.0.0-alpha.2",
+    "@jupyterlab/mathjax2-extension": "~4.0.0-alpha.2",
+    "@jupyterlab/notebook": "~4.0.0-alpha.2",
+    "@jupyterlab/notebook-extension": "~4.0.0-alpha.2",
+    "@jupyterlab/pdf-extension": "~4.0.0-alpha.2",
+    "@jupyterlab/rendermime": "~4.0.0-alpha.2",
+    "@jupyterlab/rendermime-extension": "~4.0.0-alpha.2",
+    "@jupyterlab/rendermime-interfaces": "~4.0.0-alpha.2",
+    "@jupyterlab/services": "~7.0.0-alpha.2",
+    "@jupyterlab/settingeditor": "~4.0.0-alpha.2",
+    "@jupyterlab/settingeditor-extension": "~4.0.0-alpha.2",
+    "@jupyterlab/settingregistry": "~4.0.0-alpha.2",
+    "@jupyterlab/shortcuts-extension": "~4.0.0-alpha.2",
+    "@jupyterlab/statedb": "~4.0.0-alpha.2",
+    "@jupyterlab/statusbar": "~4.0.0-alpha.2",
+    "@jupyterlab/statusbar-extension": "~4.0.0-alpha.2",
+    "@jupyterlab/terminal": "~4.0.0-alpha.2",
+    "@jupyterlab/terminal-extension": "~4.0.0-alpha.2",
+    "@jupyterlab/theme-light-extension": "~4.0.0-alpha.2",
+    "@jupyterlab/toc-extension": "~6.0.0-alpha.2",
+    "@jupyterlab/tooltip": "~4.0.0-alpha.2",
+    "@jupyterlab/tooltip-extension": "~4.0.0-alpha.2",
+    "@jupyterlab/translation": "~4.0.0-alpha.2",
+    "@jupyterlab/translation-extension": "~4.0.0-alpha.2",
+    "@jupyterlab/ui-components": "~4.0.0-alpha.17",
+    "@jupyterlab/ui-components-extension": "~4.0.0-alpha.2",
+    "@jupyterlab/vdom-extension": "~4.0.0-alpha.2",
+    "@jupyterlab/vega5-extension": "~4.0.0-alpha.2",
+    "@lumino/algorithm": "~1.9.1",
+    "@lumino/application": "~1.27.0",
+    "@lumino/commands": "~1.19.0",
+    "@lumino/coreutils": "~1.11.1",
+    "@lumino/disposable": "~1.10.1",
+    "@lumino/domutils": "~1.8.1",
+    "@lumino/dragdrop": "~1.13.1",
+    "@lumino/messaging": "~1.10.1",
+    "@lumino/properties": "~1.8.1",
+    "@lumino/signaling": "~1.10.1",
+    "@lumino/virtualdom": "~1.14.1",
+    "@lumino/widgets": "~1.30.0",
+    "react": "~17.0.2",
+    "react-dom": "~17.0.2"
```

Comment:
Ideally we could use the same logic as in `ensureJupyterLab` to also ensure the `singletonPackages` of the federated example too, as it appears some are missing.

Probably this could be defered to https://github.com/jupyterlab/jupyterlab/pull/11693, which could add utility functions to ensure the integrity of a `package.json` for a custom JupyterLab application.